source/connection-troubleshooting.txt

```diff
@@ -0,0 +1,277 @@
+
+.. _java-connection-string-port:
+
+Check Connection String
+-----------------------
+
+Verify the hostname and port number in the connection string are both
+accurate. In the sample error message, the hostname is ``127.0.0.1`` and the
+port is ``27017``. The default port value for a MongoDB instance is
+``27017``, but you can configure MongoDB to communicate on another port.
+
+Authentication Error
+~~~~~~~~~~~~~~~~~~~~
+
+When attempting to connect to a MongoDB instance, the connection can fail if
+the authorization is not correctly configured. This often results in an error
+message similar to the following:
+
+.. code-block:: none
+   :copyable: false
+
+   Command failed with error 18 (AuthenticationFailed): 'Authentication failed.' on server localhost:27017.
+
+If you receive this error, try the following methods to resolve this issue.
+
+.. _java-connection-string-auth:
+
+Check Connection String
+-----------------------
+
+The most common issue when attempting to connect to MongoDB is an invalid
+connection string. For more information about using connection
+strings with the Java driver, see :ref:`Connection URI <connection-uri>` in the
+Connection Guide.
+
+If your MongoDB deployment requires authentication, your connection string
+might require including a username and password.
+
+.. note::
+
+   If the username or password includes the following characters:
+
+   .. code-block:: none
+
+      : / ? # [ ] @
+
+   those characters must be converted using `percent encoding
+   <https://tools.ietf.org/html/rfc3986#section-2.1>`__.
+
+If your MongoDB deployment is on MongoDB Atlas, you can check your connection
+string by using the :ref:`Connect to MongoDB Atlas <connect-atlas-java-driver>`
+code example. Make sure to replace the connection string in the example.
+
+When connecting to a replica set, you should include all of the hosts
+in the replica set in your connection string. Separate each of the hosts
+in the connection string with a comma. This enables the driver to establish a
+connection if one of the hosts is unreachable.
+
+.. _java-connection-admin:
+
+Verify User is in ``admin`` Database
+------------------------------------
+
+When connecting with authentication, you need to specify the username and
+password when attempting to connect to the MongoDB deployment. If you have
+verified that the connection string is accurate, your user might not be defined
+in the ``admin`` database.
+
+You can create users for each database in your collection. By default,
+MongoDB uses the ``admin`` database for authentication. If your user is defined
+in a different database, the driver will return an error when attempting to
+connect.
+
+You could also change the ``authSource`` in the connection string to point to
+the database where the user is defined.
+
+Error Sending Message
+~~~~~~~~~~~~~~~~~~~~~
+
+When the driver is unable to send a message, it often responds with the
```

Comment:
This is one of those weird cases where this is our catch all error message. However, there are other more specific error messages that could be sent if the driver errors out when sending a command. So, there are specific cases where it wouldn't, but they are too specific and low-level to get into. That's why I went with often, because it is accurate but is not always the case.